cmd/kubeadm/app/node/csr.go

```diff
@@ -37,11 +39,16 @@ func PerformTLSBootstrap(s *kubeadmapi.KubeadmConfig, apiEndpoint string, caCert
 	// TODO(phase1+) try all the api servers until we find one that works
 	bareClientConfig := kubeadmutil.CreateBasicClientConfig("kubernetes", apiEndpoint, caCert)
 
-	nodeName, err := os.Hostname()
+	hostName, err := os.Hostname()
 	if err != nil {
 		return nil, fmt.Errorf("<node/csr> failed to get node hostname [%v]", err)
 	}
 
+	// TODO: hostname == nodename doesn't hold on all clouds (AWS).
+	// But we don't have a cloudprovider, so we're stuck.
+	glog.Errorf("assuming that hostname is the same as NodeName")
```

Comment:
I am not sure why you need to add this here, we need to open an issue to discuss this. My concern is that user has really no idea what the hell `hostname` vs `NodeName` means. I'll open a PR and an issue to fix this.